test/Microsoft.Extensions.Primitives.Tests/StringSegmentTest.cs

```diff
@@ -432,5 +444,30 @@ public void StringSegment_CreateEmptySegment()
             // Assert
             Assert.True(segment.HasValue);
         }
+
+        [Theory]
+        [InlineData("   value", 0, 8, "value")]
+        [InlineData("value   ", 0, 8, "value")]
+        [InlineData("\t\tvalue", 0, 7, "value")]
+        [InlineData("value\t\t", 0, 7, "value")]
+        [InlineData("\t\tvalue \t a", 1, 8, "value")]
+        [InlineData("   a     ", 0, 9, "a")]
+        [InlineData("value\t value  value ", 2, 13, "lue\t value  v")]
+        [InlineData("\x0009value \x0085", 0, 8, "value")]
+        [InlineData("      ", 1, 2, "")]
+        [InlineData("\t\t\t", 0, 3, "")]
+        [InlineData("\n\n\t\t  \t", 2, 3, "")]
```

Comment:
Suggest checking a few non-Latin1 whitespace characters to ensure `string` semantics are shining through.